bundles/org.eclipse.jface/src/org/eclipse/jface/resource/JFaceResources.java

```diff
@@ -583,6 +584,23 @@ public static void setFontRegistry(FontRegistry registry) {
 		fontRegistry = registry;
 	}
 
+
+	/**
+	 * Creates a local registry that wraps the ResourceManager for the current
+	 * display. Anything allocated by this registry will be automatically cleaned up
+	 * with the given control is disposed. Note that registries created in this way
+	 * should not be used to allocate any resource that must outlive the given
+	 * control.
+	 *
+	 * shortcut for
+	 * <code>LocalResourceManager(JFaceResources.getResources(), owner)</code>
+	 *
+	 * @param owner control whose disposal will trigger cleanup of everything in the
+	 *              registry.
+	 */
+	static LocalResourceManager managerFor(Control owner) {
+		return new LocalResourceManager(getResources(), owner);
```

Comment:
I would use control.getData/setData to cache the instance created so multiple calls could be performed without creating multiple managers.

Reply:
Can you estimate how often that will hit the cache?
Since the local-manager is a child from the global one caching would not save SWT-resources but mainly only another map.